src/main/java/io/camunda/zeebe/bpmnassert/testengine/PartitionCommandSenderImpl.java

```diff
@@ -8,28 +8,18 @@
 
 public class PartitionCommandSenderImpl implements PartitionCommandSender {
 
-  private final Map<Integer, SubscriptionCommandMessageHandler> subscriptionHandlers;
+  private final SubscriptionCommandMessageHandler subscriptionHandler;
 
-  public PartitionCommandSenderImpl(
-      final Map<Integer, SubscriptionCommandMessageHandler> subscriptionHandlers) {
-    this.subscriptionHandlers = subscriptionHandlers;
+  public PartitionCommandSenderImpl(final SubscriptionCommandMessageHandler subscriptionHandler) {
+    this.subscriptionHandler = subscriptionHandler;
   }
 
   @Override
   public boolean sendCommand(final int receiverPartitionId, final BufferWriter command) {
     final byte[] bytes = new byte[command.getLength()];
     final UnsafeBuffer commandBuffer = new UnsafeBuffer(bytes);
     command.write(commandBuffer, 0);
-
-    // delegate the command to the subscription handler of the receiver partition
-    try {
-      subscriptionHandlers.get(receiverPartitionId).apply(bytes);
-    } catch (NullPointerException npe) {
-      throw new RuntimeException(
-          String.format(
-              "No message subscription handler found for partition `%s`", receiverPartitionId));
-    }
-
+    subscriptionHandler.apply(bytes);
```

Comment:
:wrench: maybe add a check to this method in the beginning that the partition is what we expect it to be, and throw an exception otherwise. Without that I feel it is not clear that we are doing a deliberate modification here.

Dito in other places where it applies